tools/bulk/scripts.js

```diff
@@ -27,17 +27,223 @@ function extractOrgSite(url) {
   return { org, site };
 }
 
+/**
+ * Show a confirmation dialog with sanitization warnings
+ * @param {Object} changes - Sanitization analysis results
+ * @param {Array<{original: string, reason: string}>} changes.rejected
+ *   URLs that failed validation
+ * @param {Array<{original: string, sanitized: string, changes: string[]}>} changes.modified
+ *   URLs that were sanitized
+ * @param {string[]} changes.deduplicated - Duplicate URLs that will be removed
+ * @returns {Promise<boolean>} True if user confirms to proceed, false if cancelled
+ */
+const showSanitizationWarning = (changes) => {
+  const { rejected, modified, deduplicated } = changes;
+
+  return new Promise((resolve) => {
+    const modal = document.createElement('dialog');
+    modal.className = 'sanitization-warning';
+
+    let html = '<h2>URL Sanitization Warning</h2>';
+
+    if (rejected.length > 0) {
+      html += `<h3>⚠️ ${rejected.length} URL(s) will be rejected:</h3>`;
+      html += '<ul class="url-list rejected">';
+      rejected.forEach(({ original, reason }) => {
+        html += `<li><code>${original}</code> <span class="reason">(${reason})</span></li>`;
```

Comment:
I think this creates a possible XSS vulnerability with a maliciously constructed URL. probably need to escape the HTML or use text content (I think simple config editor has an example of HTML escaping)